### DISCUSSION TITLE
MAE MFE of trades

### DISCUSSION THREAD

**Discussion by kaustv-datta** [edtechre/pybroker#100] [Q&A] [answered]
Thanks for the very useful library!

Is it possible to extract MAE (Maximum Adverse Excursion) and MFE (Maximum Favorable Excursion) data for each trade?

**Comment by edtechre** [comment, marked as answer]
Hi @kaustv-datta,

The framework does not calculate this for you. However, I can add these metrics.

For the time being, you can calculate these yourself on the [data returned from the backtest result](https://www.pybroker.com/en/latest/notebooks/2.%20Backtesting%20a%20Strategy.html#Running-a-Backtest).

**Reply by kaustv-datta** [reply]
It would be amazing if you could add these metrics to the framework. It would help me with analyzing optimal stop-loss/take-profit levels.

**Reply by edtechre** [reply]
Hi @kaustv-datta,

Could you provide more information on what you are looking for? Do you want these computed per-trade?

Right now, the framework will return [largest_win](https://www.pybroker.com/en/latest/reference/pybroker.eval.html#pybroker.eval.EvalMetrics.largest_win) and [largest_loss](https://www.pybroker.com/en/latest/reference/pybroker.eval.html#pybroker.eval.EvalMetrics.largest_win) metrics.

**Reply by kaustv-datta** [reply]
Hi @edtechre,

I'm indeed looking for the metrics computed per-trade.

E.g. [How To Use Maximum Adverse Excursion To Optimize Your Stop Loss](https://youtu.be/BbbJNt7dheg?si=n9yZx1bTcbizVd10)

<img width="973" alt="image" src="https://github.com/edtechre/pybroker/assets/6747425/44bd4a5a-4f74-41ec-8c1f-a656402fde6e">

**Reply by edtechre** [reply]
Hi @kaustv-datta,

MAE and MFE have been added to trades in the dev branch. Please take a look! These will make it into the next v1.1.33 release.

**Reply by edtechre** [reply]
Change is in v1.1.33
